icu4j/main/core/src/main/java/com/ibm/icu/segmenter/LocalizedSegmenter.java

```diff
@@ -0,0 +1,188 @@
+// © 2025 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.segmenter;
+
+import com.ibm.icu.text.BreakIterator;
+import com.ibm.icu.util.ULocale;
+import java.util.Locale;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Performs segmentation according to the rules defined for the locale.
+ */
+public class LocalizedSegmenter implements Segmenter {
+
+  private ULocale locale;
+
+  private SegmentationType segmentationType;
+
+  @Override
+  public Segments segment(CharSequence s) {
+    return new LocalizedSegments(s, this);
+  }
+
+  /**
+   * @return a builder for constructing {@code LocalizedSegmenter}
+   * @draft ICU 78
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  LocalizedSegmenter(ULocale locale, SegmentationType segmentationType) {
```

Comment:
please try to make as many "package" methods & types private as possible -- this one for example should work

Reply:
Done.